javascript/session.ts

```diff
@@ -17,14 +17,15 @@ export class Session implements api.ISession {
     incoming: util.queue<api.IChannel>;
     enc: codec.Encoder;
     dec: codec.Decoder;
+    done: Promise<void>;
```

Comment:
I added this so that the tests could `await` the session loop. Otherwise there's an error that bubbles up in the background from `loop` that I couldn't catch.
Though maybe `close()` should `await` the loop instead of exposing this?

Reply:
i guess this isn't much different than expose a go-like wait()